tensorboard/webapp/runs/views/runs_table/runs_table_test.ts

```diff
@@ -1278,6 +1278,75 @@ describe('runs_table', () => {
       });
     });
 
+    it('filters by original experiment name and legacy run name', () => {
+      selectSpy
+        .withArgs(getExperiment, {experimentId: 'rowling'})
+        .and.returnValue(
+          of(
+            buildExperiment({
+              name: 'Harry Potter',
+            })
+          )
+        );
+
+      selectSpy
+        .withArgs(TEST_ONLY.getRunsLoading, jasmine.any)
+        .and.returnValue(of(false));
+      selectSpy
+        .withArgs(getRuns, {experimentId: 'rowling'})
+        .and.returnValue(
+          of([
+            buildRun({id: 'book1', name: "The Philosopher's Stone"}),
+            buildRun({id: 'book2', name: 'The Chamber Of Secrets'}),
+          ])
+        );
+
+      selectSpy
+        .withArgs(getExperiment, {experimentId: 'tolkien'})
+        .and.returnValue(
+          of(
+            buildExperiment({
+              name: 'The Lord of the Rings',
+            })
+          )
+        );
+      selectSpy
+        .withArgs(getRuns, {experimentId: 'tolkien'})
+        .and.returnValue(
+          of([
+            buildRun({id: 'book3', name: 'The Fellowship of the Ring'}),
+            buildRun({id: 'book4', name: 'The Silmarillion'}),
+          ])
+        );
+
+      store.overrideSelector(getExperimentIdToAliasMap, {
+        rowling: 'HPz',
+        tolkien: 'LotR',
+      });
+      store.overrideSelector(getRunSelectorRegexFilter, 'ings');
+
+      const fixture = createComponent(
+        ['rowling', 'tolkien'],
+        [RunsTableColumn.EXPERIMENT_NAME, RunsTableColumn.RUN_NAME]
+      );
+
+      fixture.detectChanges();
+      expect(getTableRowTextContent(fixture)).toEqual([
+        ['LotR', 'The Fellowship of the Ring'],
+        ['LotR', 'The Silmarillion'],
+      ]);
+
+      // Alias for Harry Potter contains "z". If we were matching against
+      // alias + run name, Harry Potter should not match below.
+      store.overrideSelector(getRunSelectorRegexFilter, 'o[^z]+/.+S[ei]');
```

Comment:
Despite the comment, I found myself having to jump around a bunch to realize that the reason "The Chamber Of Secrets" matches because this regex here matches "otter/The Chamber Of Se".  imo this would be more readable if we didn't use control characters in this regex.

Reply:
For now, I left a comment on what the regex is expecting instead of changing the regex itself. Hopefully no one will get too confused by this test.